### PR TITLE
chore: add suggestion on class naming

### PR DESCRIPTION
**Pull request by Derollez**
- `LinearInterpolator` > `Linear` to follow the patterns on other objects.

### CODE REVIEW

File: include/OpenSpaceToolkit/Mathematics/CurveFitting/Interpolation/Linear.hpp [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-/// @author         Vishwa Shah <vishwa@loftorbital.com
+/// @file           OpenSpaceToolkit/Mathematics/Interpolation/Linear.hpp
+/// @author         Vishwa Shah <vishwa@loftorbital.com>
 /// @license        Apache License 2.0
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#ifndef __OpenSpaceToolkit_Mathematics_Interpolator_LinearInterpolator__
-#define __OpenSpaceToolkit_Mathematics_Interpolator_LinearInterpolator__
+#ifndef __OpenSpaceToolkit_Mathematics_Interpolator_Linear__
```
Comment (lucas-bremond) [inline include/OpenSpaceToolkit/Mathematics/CurveFitting/Interpolation/Linear.hpp:10]: Folder is `Interpolation`, but path here refers to`Interpolator`.

File: include/OpenSpaceToolkit/Mathematics/CurveFitting/Interpolator.hpp [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 /// @project        Open Space Toolkit ▸ Mathematics
 /// @file           OpenSpaceToolkit/Mathematics/Interpolation/CubicSpline.hpp
```
Comment (lucas-bremond) [inline include/OpenSpaceToolkit/Mathematics/CurveFitting/Interpolator.hpp:4]: Doesn't match file name?

**Comment by lucas-bremond** [comment]
Overall, I believe the decision should be between having an `Interpolation` directory, and keeping `XXXInterpolator` in the class names; or having an `Interpolator` directory, and removing it.

**Review by vishwa2710 — DISMISSED** [review]
Updated the naming as appropriate.

**Review by vishwa2710 — APPROVED** [review]

**Pull request merged**